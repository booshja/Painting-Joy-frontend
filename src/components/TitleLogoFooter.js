// dependencies
import React from "react";
import styled from "styled-components";
// components
import { Link } from "react-router-dom";
// assets
import Logo from "../assets/painting-joy-logo-white.jpg";

const StyledTitleLogo = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
`;

const StyledLogo = styled.img`
    width: 150px;
    height: 150px;
`;

const TitleLogoFooter = () => {
    return (
        <StyledTitleLogo>
            <Link to="/">
                <StyledLogo src={Logo} alt="Painting Joy Mural Co. Logo" />
            </Link>
        </StyledTitleLogo>
    );
};

export default TitleLogoFooter;
