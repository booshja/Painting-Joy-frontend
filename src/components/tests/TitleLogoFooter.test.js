import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import TitleLogoFooter from "../TitleLogoFooter";
import "jest-styled-components";

/** Smoke Test */
it("renders without crashing", () => {
    render(
        <BrowserRouter>
            <TitleLogoFooter />
        </BrowserRouter>
    );
});

/** Snapshot Test */
it("matches snapshot", () => {
    const { asFragment } = render(
        <BrowserRouter>
            <TitleLogoFooter />
        </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

/** RTL Tests */
it("renders correctly", () => {
    const { getByAltText } = render(
        <BrowserRouter>
            <TitleLogoFooter />
        </BrowserRouter>
    );

    expect(getByAltText("Painting Joy Mural Co. Logo")).toBeInTheDocument();
});
