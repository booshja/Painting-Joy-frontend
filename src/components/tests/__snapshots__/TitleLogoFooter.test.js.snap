// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  width: 150px;
  height: 150px;
}

<div
    class="c0"
  >
    <a
      href="/"
    >
      <img
        alt="Painting Joy Mural Co. Logo"
        class="c1"
        src="painting-joy-logo-white.jpg"
      />
    </a>
  </div>
</DocumentFragment>
`;
